pkg/infradb/taskmanager/taskmanager_test.go

```diff
@@ -0,0 +1,295 @@
+}
+func handleTestEventbusy(wg *sync.WaitGroup) {
+	name := "testTaskbusy"
+	objectType := "testTypebusy"
+	resourceVersion := "testVersionbusy"
+	dropTask := false
+
+	component := &common.Component{
+		CompStatus: common.ComponentStatusSuccess,
+	}
+
+	TaskMan.StatusUpdated(name, objectType, resourceVersion, "notificationID", dropTask, component)
+
+	retValMu.Lock()
+	retVal = true
+	retValMu.Unlock()
+
+	wg.Done()
+}
+func handletestEventTimeout() {
+
+}
+func handletestNotificationIDNotMatching(objectData *eventbus.ObjectData) {
+	name := "testTaskNotificationIdNotMatching"
+	objectType := "testTypeNotificationIdNotMatching"
+	resourceVersion := "testVersionNotificationIdNotMatching"
+	dropTask := false
+
+	component := &common.Component{
+		Name:       "testModuleNotificationIdNotMatching",
+		CompStatus: common.ComponentStatusSuccess,
+	}
+	TaskMan.StatusUpdated(name, objectType, resourceVersion, "NotificationIdNotMatching", dropTask, component)
+
+	time.Sleep(100 * time.Millisecond)
+
+	TaskMan.StatusUpdated(name, objectType, resourceVersion, objectData.NotificationID, dropTask, component)
+}
+func handleTestEventError(objectData *eventbus.ObjectData) {
+	name := "testTaskEventError"
+	objectType := "testTypeEventError"
+	resourceVersion := "testVersionEventError"
+
+	dropTask := false
+
+	component := &common.Component{
+		CompStatus: common.ComponentStatusError,
+	}
+	if component.Timer == 0 {
+		component.Timer = 2 * time.Second
+	} else {
+		component.Timer *= 2
+	}
+	TaskMan.StatusUpdated(name, objectType, resourceVersion, objectData.NotificationID, dropTask, component)
+}
+func (h *ModulelciHandler) HandleEvent(eventType string, objectData *eventbus.ObjectData) {
+	h.receivedEvents = append(h.receivedEvents, objectData)
+	switch eventType {
+	case "testEvent":
+		handleTestEvent(objectData)
+	case "testEventCompSuccess":
+		handleTestEventCompSuccess(objectData)
+	case "testEventError":
+		handleTestEventError(objectData)
+	case "testEventTimeout":
+		handletestEventTimeout()
+	case "testEventNotificationIdNotMatching":
+		handletestNotificationIDNotMatching(objectData)
+	default:
+		log.Printf("LCI: error: Unknown event type %s", eventType)
+	}
+}
+
+func TestTaskManager_CreateTask(t *testing.T) {
+	subscriber := &eventbus.Subscriber{
+		Name:     "testSubscriber",
+		Ch:       make(chan interface{}),
+		Quit:     make(chan bool),
+		Priority: 1,
+	}
+	subs := []*eventbus.Subscriber{subscriber}
+
+	TaskMan.StartTaskManager()
+	TaskMan.CreateTask("testTask", "testType", "testVersion", subs)
+
+	time.Sleep(100 * time.Millisecond)
+
+	task := TaskMan.taskQueue.Dequeue()
+	assert.NotNil(t, task)
+	assert.Equal(t, "testTask", task.name)
+	assert.Equal(t, "testType", task.objectType)
+	assert.Equal(t, "testVersion", task.resourceVersion)
+	assert.Equal(t, subs, task.subs)
+}
+
+func TestTaskManager_CompSuccess(t *testing.T) {
+	var wg sync.WaitGroup
+
+	TaskMan.StartTaskManager()
+	handler := &ModulelciHandler{}
+	wg.Add(1)
```

Comment:
maybe you want to have this wg within StartSubscriber, so it exits only when we are listening for event?

Reply:
then we have to change the whole design. which will impact all the other modules/pkgs also ?